deployment/helm/templates/agent.yaml

```diff
@@ -103,7 +103,17 @@ spec:
           path: "{{ .Values.agent.host.kubeletDevicePlugins }}"
       - name: var-run-dockershim
         hostPath:
-          path: "{{ .Values.agent.host.dockerShimSock }}"
+          {{- if ne "" .Values.agent.host.containerRuntimeSocket }}
+          path: {{.Values.agent.host.containerRuntimeSocket }}
+          {{- else if eq .Values.kubernetesDistro "microk8s" }}
+          path: "/var/snap/microk8s/common/run/containerd.sock"
+          {{- else if eq .Values.kubernetesDistro "k3s" }}
+          path: "/run/k3s/containerd/containerd.sock"
+          {{- else if eq .Values.kubernetesDistro "k8s" }}
+          path: "/var/run/dockershim.sock"
```

Comment:
What if folks use containerd instead of dockershim?

Reply:
then they can set `agent.host.containerRuntimeSocket` to whatever they want

Reply:
The change in documentation should clarify that https://github.com/project-akri/akri-docs/pull/19

Reply:
Would it make sense to then use k8s-dockershim to reduce confusion? Otherwise folks with containerd might be surprised why it is not working. Just FYI, AKS has transitioned from dockershim to containerd, the future lies with containerd, not dockershim.

Reply:
in that case i think it would be better to do the alternative i describe in the PR description. Namely, we would provide no `kubernetesDistro=$DISTRO_NAME` and instead recommend how to set `agent.host.containerRuntimeSocket` based on the distro. I think that would be better than providing multiple options per distro (k8s-containerd`, `k8s-dockershim`, etc). @bfjelds and @romoh on thoughts?

Reply:
@jiria @romoh @bfjelds bump here. Would be nice to get the documentation in one this once we decide on what to do here

Reply:
i would say have distro=k8s|k3s|etc ... then use whatever the distro default is (dockershim or containerd) ... and document that anything that doesn't follow the defaults can be supported iwth containerRuntimeSocket.  though i also have no problem with k8s-dockershim|k8s-containerd|etc.

Reply:
I am going with brian's last comment and sticking with what we have here. If folks complain about wanting our k8s default changed, we can change it. And I'll make sure the documentation points out how to set it to containerd.

Reply:
Sounds good to me.